presto-session-property-managers/src/main/java/com/facebook/presto/session/SessionMatchSpec.java

```diff
@@ -99,7 +107,10 @@ public Map<String, String> match(SessionConfigurationContext context)
             }
         }
 
-        return sessionProperties;
+        Map<String, String> validSessionProperties = new HashMap<>();
+        sessionProperties.entrySet().stream().filter(property -> coordinatorVersion.lessThanOrEqualTo(maxVersion) && coordinatorVersion.greaterThanOrEqualTo(minVersion)).forEach(entry -> validSessionProperties.put(entry.getKey(), entry.getValue()));
```

Comment:
`Maps.filterEntries` from guava can do it one line.

However I'm curious what filtering is really happening here since the predicate condition is a static check and doesn't really use any key or value information.
`coordinatorVersion.lessThanOrEqualTo(maxVersion) && coordinatorVersion.greaterThanOrEqualTo(minVersion))`

Reply:
Good point. Updated the diff.